spring-boot-ecommerce/src/main/java/com/luv2code/ecommerce/service/ProductService.java

```diff
@@ -13,12 +17,31 @@ public class ProductService {
     @Autowired
     private ProductRepository productRepository;
 
-    public List<Product> fetchAllProduct(){
-        List<Product> productList=productRepository.findAll();
-        for(Product product: productList){
+       public List<Product> fetchAllProduct1(){
+           List<Product> productList=productRepository.findAll();
 
+           return productList;
+       }
+
+    public List<ProductResponseDto> fetchAllProduct(){
+        List<Product> products= productRepository.findAll();
+        // List<Demo_Product> demo_products= products.stream().map(product -> new Demo_Product(product.getId(), product.getImageUrl(), product.getName())).collect(Collectors.toList());
+        ProductResponseDto demo_products=null;
+        ArrayList<ProductResponseDto> products1= new ArrayList<>();
+        for (Product product: products){
+            demo_products= new ProductResponseDto(product.getId(), product.getImageUrl(), product.getName());
+            products1.add(demo_products);
         }
-        return productList;
+        return products1;
+    }
+
+    public  ProductResponseDto fetchProductById(Long id){
+          Product product= productRepository.findById(id).get();
```

Comment:
use Optional<Product> and check product is null or not

Reply:
When we use Optional the we can't check the product is null or not the Sonar Lint wll thrown an exception on this

Reply:
So i will update the ProductService and also use Optional in it.